src/snowflake/snowpark/_internal/data_source/dbms_dialects/base_dialect.py

```diff
@@ -7,6 +7,9 @@
 
 
 class BaseDialect:
+    def __init__(self, is_query: bool) -> None:
+        self.is_query = is_query
```

Comment:
from OO design perspective, it feels a bit weird to me that the Dialect instance has a is_query mem var.

can we move this flag to the static methods?